### DISCUSSION TITLE
Multiple scroll strips

### DISCUSSION THREAD

**Discussion by Aqa-Ib** [YaLTeR/niri#458] [Ideas]
I wonder what would be the complexity required for having multiple scroll strips. For example, one on top of the other, having the ability to be sized independently as a whole.

Do you imagine this being easy or difficult to implement and maintain ?

Thanks!

**Comment by YaLTeR** [comment]
Would need quite a bit of refactoring and doesn't sound like it adds enough over the workspaces

**Reply by Aqa-Ib** [reply]
Well, this would be useful in two situations: 

- When using a screen larger than 40 inches, it **needs** to have two strips one on top of the other.
- If one day niri gains the ability to have vertical scrollable strips, a normal sized screen can easily hold two of them side by side.

Thanks for your response and wish you the best !

**Reply by YaLTeR** [reply]
Hm, that is quite a screen size. Yeah, I agree it could be useful then.

However, I also right away see a design issue: currently fullscreen windows act as a regular column, but what do you do when you have two strips?

**Comment by salman-farooq-sh** [comment]
Will the immplementation be simpler if we conceptualize the strips just as multiple workspaces visible at once. (with an integer or maybe even a float (if we are being fancy) config variable to specify how many are visible on a screen at once).

OR

We can treat a single monitor as multiple divided-up ones, like the built-in picture-by-picture feature of many monitors (I think that's what it's called but I can be wrong). 

But the problem with the thing built in the monitor is that:
1.You need such a monitor
2. I have only seen it to be pretty limited in options e.g. my 32" Philips 4K can only divide into two halves side-by-side, not up and down. A 40" one might give more options.
3. Takes two display ports on your machine.
4. Fullscreen-ing as in ignoring the division is a problem.

**Reply by YaLTeR** [reply]
> We can treat a single monitor as multiple divided-up ones

That might be simpler to implement; though the fullscreen problem is still there. With a subdivision like this making a window fullscreen will make it cover half of the monitor. Which is correct for this particular way of subdividing, but not necessarily what you want.

**Reply by salman-farooq-sh** [reply]
One to all of the following can be supported leaving the choice on user, including the one for the default:
1. Fullscreen-ing takes up the whole division only
2. "Fake" fullscreen, will need to have at least one of height or width or ratio for this.
3. Fullscreen-ing takes up the whole monitor

**Comment by salman-farooq-sh** [comment]
Possible inexact solution for: https://github.com/YaLTeR/niri/discussions/450
